src/jltypes.c

```diff
@@ -2670,6 +2726,8 @@ int jl_subtype_le(jl_value_t *a, jl_value_t *b, int ta, int invariant)
                 return 0;
             }
             tta = tta->super; super = 1;
+            if (tta == NULL)
+                jl_error("cannot compute subtype relation for type under construction");
```

Comment:
If you want to work on that later, I could also just return 0 here (like it did previously) and let the caller throw the error